LCM/dsc/engine/ca/CAInfrastructure/WebPullClient.c

```diff
@@ -2875,13 +2901,20 @@ MI_Result MI_CALL Pull_SendStatusReport(_In_ LCMProviderContext *lcmContext,
         curl_easy_setopt(curl, CURLOPT_HEADERDATA, &headerChunk);
         curl_easy_setopt(curl, CURLOPT_WRITEDATA, &dataChunk);
 
-        curl_easy_setopt(curl, CURLOPT_SSLCERT, OAAS_CERTPATH);
+        res = curl_easy_setopt(curl, CURLOPT_SSLCERT, OAAS_CERTPATH);
+        if (res != CURLE_OK)
+        {
+            curl_slist_free_all(list);
+            curl_easy_cleanup(curl);
+            return GetCimMIError(MI_RESULT_FAILED, extendedError, ID_PULL_CERTOPTS_NOT_SUPPORTED);
+        }
         curl_easy_setopt(curl, CURLOPT_SSLKEY, OAAS_KEYPATH);
         
         res = curl_easy_perform(curl);
 
         if (res != CURLE_OK)
         {
+            GetCimMIError2Params(MI_RESULT_FAILED, extendedError, ID_PULL_CURLPERFORMFAILED, curl, curl_easy_strerror(res));
```

Comment:
I'm not sure that this function actually writes to the log. It usually just retrieves the error string with the parameters replaced that we return to throw an error with the correct message. This string would need to be returned to actually throw an error and print to the log. Not sure what it needs to just write to the log without throwing an error. Can you confirm that this function will actually print to the log by itself?

Also this code will still hit the un-initialized responseCode variable in the error message at the bottom of this function if an error occurs here, so that variable needs to be populated in this case or that error needs to be changed.

Reply:
Yes this does write to our log. All this function does is format an MI error string and then log it and return it. Since we don't want to throw an error in this case, I'm not doing anything with the return value. Here's where this function is defined: https://github.com/Microsoft/PowerShell-DSC-for-Linux/blob/master/LCM/dsc/engine/EngineHelper/EngineHelper.c#L193

Reply:
Ok. The un-initialized responseCode variable still needs to be handled though